pkg/storage/registry/spaces/spaces.go

```diff
@@ -172,43 +184,43 @@ type registry struct {
 	// the template to use when determining the home provider
 	homeTemplate *template.Template
 	// a map of resources to providers
-	resources map[string][]*registrypb.ProviderInfo
-	// a map of paths/aliases to spaces and providers
-	// aliases           map[string]map[string]*spaceAndProvider
+	resources         map[string][]*registrypb.ProviderInfo
 	resourceNameCache map[string]string
 
 	getStorageProviderServiceClient GetStorageProviderServiceClientFunc
 }
 
 // GetProvider return the storage provider for the given spaces according to the rule configuration
 func (r *registry) GetProvider(ctx context.Context, space *providerpb.StorageSpace) (*registrypb.ProviderInfo, error) {
-	for address, rule := range r.c.Providers {
-		mountPath := ""
-		var err error
-		if space.SpaceType != "" && rule.SpaceType != space.SpaceType {
-			continue
-		}
-		if space.Owner != nil {
-			mountPath, err = rule.ProviderPath(nil, space)
-			if err != nil {
+	for address, provider := range r.c.Providers {
+		for spaceType, sc := range provider.Spaces {
+			spacePath := ""
+			var err error
+			if space.SpaceType != "" && spaceType != space.SpaceType {
 				continue
 			}
-			match, err := regexp.MatchString(rule.MountPath, mountPath)
-			if err != nil {
-				continue
+			if space.Owner != nil {
+				spacePath, err = sc.SpacePath(nil, space)
```

Comment:
Shouldn't we pass `space.Owner` instead `nil` here? We explicitly checked it is not nil?

Reply:
the first parameter is the current user, however for personal spaces the current user may not be the owner of a space. The latter is part of the space struct we pass as the second parameter to `SpacePath`.